docs/DeveloperGuide.md

```diff
@@ -38,6 +38,69 @@ title: Developer Guide
 - Edit tags - change category of spending
 - Remove tags - delete category of spending
 
+
+## Use Cases
+The following documents use cases for our application
+
+For the following Use Cases (unless specified otherwise):
+- The System is `UniCa$h`
+- The Actor is `User`
+
+#### Use Case: UC01 - Adding an income
+**MSS:**
+1. User enters the command to add an income with the correct format.
+2. User submits the request.
+3. UniCa$h adds the income to the income list and displays success message.
+    
+    Use Case ends
+
+**Extensions**
+- 2a. User enters an incorrect format
+  - 2a1. UniCa$h displays an error message.
+  - Use case resumes at step 1.
+- 2b. User enters an extremely high number
+  - 2b1. UniCa$h prompts the user with a warning of the input
+  - 2b2. User confirms the request
+  - Use case resumes at step 3.
+
+#### Use Case: UC02 - Finding an income
+**MSS:**
+1. User enters the command to find an income with the correct format.
+2. User submits the request.
+3. UniCa$h filters all the incomes based on the specified filters and returns
+the filtered income with a success message.
+
+   Use Case ends
+
+**Extensions**
+- 2a. User enters an incorrect format.
+  - 1a1. UniCa$h displays an error message.
+  - Use case resumes at step 1.
+
+- 3a. UniCa$h does not find any results matching the filter.
+  - 3a1. UniCa$h displays a message saying no results found.
+  - Use Case resumes at step 1
+
+#### Use Case: UC03 - Delete an income
+**MSS:**
+1. User enters the command to delete an income with the correct format.
+2. User submits the request.
+3. UniCa$h finds the income based on the passed in arguments.
+4. UniCa$h deletes the income.
+5. UniCa$h displays success message.
+
+   Use Case ends
+
+**Extensions**
+- 2a. User enters an incorrect format.
+    - 1a1. UniCa$h displays an error message.
```

Comment:
Same suggestion as mentioned previously. Would it also be better to specify that the UniCa$h will request for the correct format?